plugins/transport-reactor-netty4/src/main/java/org/opensearch/http/reactor/netty4/ReactorNetty4HttpChunk.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.http.reactor.netty4;
+
+import org.opensearch.core.common.bytes.BytesReference;
+import org.opensearch.http.HttpChunk;
+import org.opensearch.transport.reactor.netty4.Netty4Utils;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import io.netty.buffer.ByteBuf;
+
+class ReactorNetty4HttpChunk implements HttpChunk {
+    private final AtomicBoolean released;
+    private final boolean pooled;
+    private final ByteBuf content;
+    private final boolean last;
+
+    ReactorNetty4HttpChunk(ByteBuf content, boolean last) {
+        this(new AtomicBoolean(false), true, content, last);
+    }
+
+    private ReactorNetty4HttpChunk(AtomicBoolean released, boolean pooled, ByteBuf content, boolean last) {
+        this.content = content;
+        this.pooled = pooled;
+        this.released = released;
+        this.last = last;
+    }
+
+    @Override
+    public BytesReference content() {
+        assert released.get() == false;
```

Comment:
Any reason not to do the following?

```suggestion
        if (released.get()) {
            throw new IllegalStateException();
        }
```

Reply:
This one follows the conventions from `Netty4HttpRequest` / `ReactorNetty4HttpRequest` with respect to release impl